### PR TITLE
chore: add editorconfig to enforce private and public member naming convention 

### PR DESCRIPTION
**Pull request by daniel-pebble**
### What this PR does / why we need it
This PR adds a editorconfig file which will enforce good naming convention when using support IDEs.  Thankfully most private/public members already conform to this standard. 
Private = _camalCase
Public = PascalCase

### Which issue(s) this PR resolves / fixes
N/A

### Please check the following list
- [ ] Does the affected code have corresponding tests, e.g. unit test, E2E test? No
- [ ] Does this change require a documentation update? No
- [ ] Does this introduce breaking changes that would require an announcement or bumping the major version? No
- [ ] Do all new files have an appropriate license header? No

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+#### Core EditorConfig Options ####
+
+# Indentation and spacing
+indent_size = 4
+indent_style = space
+tab_width = 4
+
+# New line preferences
+end_of_line = crlf
```
Comment (shizhMSFT) [inline .editorconfig:28]: All code in this repo is currently `lf`

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+insert_final_newline = false
```
Comment (shizhMSFT) [inline .editorconfig:29]: We do insert final newline.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**